mlflow/projects/__init__.py

```diff
@@ -322,6 +322,8 @@ def _get_conda_bin_executable(executable_name):
     ``mlflow.projects.MLFLOW_CONDA_HOME`` is unspecified, this method simply returns the passed-in
     executable name.
     """
+    if MLFLOW_CONDA_HOME != "MLFLOW_CONDA_HOME":
```

Comment:
This would be a no-op since `MLFLOW_CONDA_HOME` constant is set to `"MLFLOW_CONDA_HOME"` string @ [line 45](https://github.com/mlflow/mlflow/blob/03f9c82e3ed2915e2dfc6cb2d32ee49c04be84f2/mlflow/projects/__init__.py#L45)

Reply:
not if the user does : 

`mlflow.projects.MLFLOW_CONDA_HOME = "path/to/anaconda.."`

it might seem more intuitive from the user's standpoint to communicate the conda path to mlflow in a way that changes a variable inside mlflow rather than going through the system's environment variables.